### ISSUE TITLE
Secure library.kiwix.org catalog part.

### ISSUE THREAD

**Issue by mgautierfr** [kiwix/kiwix-tools#397]
https://library.kiwix.org use `kiwix-serve` to provide two main types of content:

- The catalog part (opds feeds, search for zim file according the criteria, ...)
- The zim content part (main page, zim articles, fulltext search, ...).

They are both served by `kiwix-serve` (through `kiwix-lib`) but it follow to different code path:

- The catalog part mainly (if not only) works on the `library.xml` as source content. It main works is mainly to parse the request and generate the corresponding opds feed. No use of the `libzim` is made.
- On the opposite side, the content part greatly use the libzim. It is time consuming and is greatly subject to evolution of the libzim (and its potential bugs).

On top of that, the catalog part is a mandatory piece of our system as it is used by our client softwares to get the list of available zim files and the url to download them. The content part is "only" used as a demo of our technology/catalog.
(Not that the content part is not important. It is important that everything works correctly and we fix (performance) issues. But one part is mandatory (a dependency) and the other not)

Having the two part served by the same process can lead to the catalog part being break because of a issue not related to it or simply because of too much requests on the content part.

The solution I propose is to split the two part in to different processes.
It could be in two different binaries or we could keep `kiwix-serve` but launch two processes with different options. This way, if the content part crash or take too much time, the catalog part would still be available.
It would be to the proxy to forward `https://library.kiwix.org/catalog/...` requests to the catalog process and other requests to the content process.

**Comment by kelson42** [comment]
What about having just two Docker containers?

**Comment by mgautierfr** [comment]
Yes, indeed, we don't even have to change the software. Just two docker containers and the correct mapping in the proxy.

**Comment by kelson42** [comment]
@mgautierfr I will transfer the issue and we will do our best to implement it quickly
